hiero-enterprise-base/src/test/java/com/openelements/hiero/base/test/AccountClientImplTest.java

```diff
@@ -0,0 +1,98 @@
+package com.openelements.hiero.base.test;
+import com.openelements.hiero.base.implementation.AccountClientImpl;
+import com.hedera.hashgraph.sdk.Hbar;
+import com.openelements.hiero.base.HieroException;
+import com.openelements.hiero.base.protocol.AccountBalanceRequest;
+import com.openelements.hiero.base.protocol.AccountBalanceResponse;
+import com.openelements.hiero.base.protocol.ProtocolLayerClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class AccountClientImplTest {
+
+    private ProtocolLayerClient mockProtocolLayerClient;
+    private AccountClientImpl accountClientImpl;
+
+    @BeforeEach
+    public void setUp() {
+        mockProtocolLayerClient = mock(ProtocolLayerClient.class);
+        accountClientImpl = new AccountClientImpl(mockProtocolLayerClient);
+    }
+
+    @Test
+    public void testGetAccountBalance_ValidPositiveBalance() throws HieroException {
+        // Fully qualify AccountId to ensure no ambiguity
+        com.hedera.hashgraph.sdk.AccountId accountId = com.hedera.hashgraph.sdk.AccountId.fromString("0.0.12345");
```

Comment:
There is only one `AccountId` on the classpath. Based on that I would favor to do a regular import instead of a full qualified class in code.